java/src/shared/facades/DevCardFacade.java

```diff
@@ -88,7 +88,11 @@ public void buyDevCard(@NotNull Player currentPlayer) {
             bank.setOfType(t, bank.getOfType(t) - all.getOfType(t));
         }
         DevCardType randomType = bank.getRandom();
-        currentPlayer.getNewDevCards().setOfType(randomType, currentPlayer.getNewDevCards().getOfType(randomType) + 1);
+        if (randomType == DevCardType.MONUMENT) {
```

Comment:
This would be a lot cleaner without the if statement. Use a local variable to store and modify `oldDevCards` or `newDevCards` as the code inside the if statement only varies by that. Something like `devCards = randomType == DevCardType.MONUMENT ? currentPlayer.getOldDevCards() : currentPlayer.getNewDevCards()`